### PR TITLE
Added data nuget

### PR DESCRIPTION
**Pull request by BelehovEgor**
Reworked the part responsible for working with the database. It is placed in a separate nouget package, which contains an interface for working with database that no longer depends on EntityFramework, as well as its implementation of ServiceDbContext which will be injected for this interface.

### CODE REVIEW

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    /// <summary>
+    /// Get collection.
+    /// </summary>
+    IQueryable<T> Get<T>() where T : class;
+
+    /// <summary>
+    /// Save data changes.
+    /// </summary>
+    void Save();
```
Comment (MariyaSin) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:32]: add empty line

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /// <summary>
+    /// Async save data changes.
+    /// </summary>
+    Task SaveAsync();
```
Comment (MariyaSin) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:36]: same

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    /// <summary>
+    /// Detach entity.
+    /// </summary>
+    object MakeEntityDetached(object obj);
```
Comment (MariyaSin) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:40]: same

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    /// <summary>
+    /// Ensure database deleted.
+    /// </summary>
+    void EnsureDeleted();
```
Comment (MariyaSin) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:44]: same

**Review by MariyaSin — APPROVED** [review]

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+{
+  /// <summary>
+  /// Base data provider interface.
+  /// </summary>
+  [AutoInject(Enums.InjectType.Scoped)]
+  public interface IDataProvider
+  {
+    /// <summary>
+    /// Async add new item.
+    /// </summary>
```
Comment (ndoriomedov) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:16]: why async?

File: src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    void Add<T>(T item) where T : class;
+
+    /// <summary>
+    /// ASync add collection of items.
+    /// </summary>
```
Comment (ndoriomedov) [inline src/Kernel.DataSupport/Database/Interfaces/IDataProvider.cs:21]: same

**Review by ndoriomedov — APPROVED** [review]

**Pull request closed without merging**